compiler/luci/pass/src/ResolveCustomOpMaxPoolWithArgmaxPass.cpp

```diff
@@ -0,0 +1,670 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/ResolveCustomOpMaxPoolWithArgmaxPass.h"
+
+#include "flatbuffers/flexbuffers.h"
+#include <loco/IR/DataTypeTraits.h>
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <loco.h>
+#include <oops/InternalExn.h>
+
+#include <iostream>
+
+namespace
+{
+
+template <typename T> std::vector<T> to_vector(const flexbuffers::TypedVector &typed_vec)
+{
+  std::vector<T> answer(typed_vec.size());
+
+  for (uint32_t i = 0; i < answer.size(); ++i)
+  {
+    answer[i] = typed_vec[i].As<T>();
+  }
+
+  return answer;
+}
+
+void copy_shape(const luci::CircleNode *src, luci::CircleNode *dst)
+{
+  assert(src->shape_status() == luci::ShapeStatus::VALID);
+
+  dst->rank(src->rank());
+  for (uint32_t i = 0; i < dst->rank(); ++i)
+  {
+    dst->dim(i).set(src->dim(i).value());
+  }
+
+  dst->shape_status(luci::ShapeStatus::VALID);
+}
+
+luci::Padding string_to_padding(const std::string &pad_str)
+{
+  if (pad_str == "VALID")
+    return luci::Padding::VALID;
+  if (pad_str == "SAME")
+    return luci::Padding::SAME;
+
+  return luci::Padding::UNDEFINED;
+}
+
+template <loco::DataType DT> void fill_conv_weights(luci::CircleConst *weights)
+{
+  assert(weights->rank() == 4);
+
+  auto const kn = weights->dim(0).value();
+  auto const kw = weights->dim(1).value();
+  auto const kh = weights->dim(2).value();
```

Comment:
If this is NHWC and N is 0, I assume `kw` is for width, then is kw accessing dim(1) correct?